libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -299,54 +294,63 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
+        esc_telem_send_struct_fn_t fn = nullptr;
+
+        // make sure a msg hasn't been extended
+        static_assert(MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_5_TO_8_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_9_TO_12_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_13_TO_16_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_17_TO_20_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_21_TO_24_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_21_TO_24_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_25_TO_28_LEN &&
+                      MAVLINK_MSG_ID_ESC_TELEMETRY_1_TO_4_LEN == MAVLINK_MSG_ID_ESC_TELEMETRY_29_TO_32_LEN,
+                      "telem messages not compatible");
 
         // send messages
         switch (i) {
             case 0:
-                fn = mavlink_msg_esc_telemetry_1_to_4_send;
+                fn = mavlink_msg_esc_telemetry_1_to_4_send_struct;
                 break;
             case 1:
-                fn = mavlink_msg_esc_telemetry_5_to_8_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_5_to_8_send_struct;
                 break;
             case 2:
-                fn = mavlink_msg_esc_telemetry_9_to_12_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_9_to_12_send_struct;
                 break;
             case 3:
-                fn = mavlink_msg_esc_telemetry_13_to_16_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_13_to_16_send_struct;
                 break;
 #if ESC_TELEM_MAX_ESCS > 16
             case 4:
-                fn = mavlink_msg_esc_telemetry_17_to_20_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_17_to_20_send_struct;
                 break;
             case 5:
-                fn = mavlink_msg_esc_telemetry_21_to_24_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_21_to_24_send_struct;
                 break;
             case 6:
-                fn = mavlink_msg_esc_telemetry_25_to_28_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_25_to_28_send_struct;
                 break;
             case 7:
-                fn = mavlink_msg_esc_telemetry_29_to_32_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_29_to_32_send_struct;
                 break;
 #endif
         }
         if (fn != nullptr) {
-            fn((mavlink_channel_t)mav_chan, temperature, voltage, current, current_tot, rpm, count);
+            fn((mavlink_channel_t)mav_chan, &s);
```

Comment:
at this point, won't it be simplier to call the functions directly from the switch case ? the function pointer looks a bit overcomplicated for a simple call with 2 params

Reply:
it is less flash to do it this way